deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/circuitbreaker/CircuitBreakerConfigurationParser.java

```diff
@@ -0,0 +1,59 @@
+package io.quarkiverse.openapi.generator.deployment.circuitbreaker;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+import java.util.function.UnaryOperator;
+import java.util.stream.Collectors;
+
+import io.quarkiverse.openapi.generator.deployment.circuitbreaker.CircuitBreakerConfiguration.CircuitBreakerClassConfiguration;
+
+public final class CircuitBreakerConfigurationParser {
+
+    private final UnaryOperator<String> nameToValuePropertyMapper;
+
+    public CircuitBreakerConfigurationParser(UnaryOperator<String> nameToValuePropertyMapper) {
+        this.nameToValuePropertyMapper = nameToValuePropertyMapper;
```

Comment:
I think is better to have a static method that just acceps the `Config` object

`public static CircuitBreakerConfiguration of (Config config) {
return new CircuirBreakConfiguration(readClasses(config));
}
`

Reply:
I agree it would make the code much simpler, but it would also couple this class with MicroProfile Config. Besides that, it would make it harder to test, since it would need a Config.
WDYT?